### ISSUE TITLE
Map change crash

### ISSUE THREAD

**Issue by fokkonaut** [ddnet/ddnet#4274]
I have made a design changer which usually works. For some unknown reason it sometimes crashes the DDNet client here:
https://github.com/ddnet/ddnet/blob/2b1ff1c823337fa3febb0a782334df05b9870c3b/src/game/client/gameclient.cpp#L2231

The crash is in GetTile where it tries to access m_pTiles with an invalid index somehow.

**Comment by def-** [comment]
Can you easily run the client with asan? That will probably tell you a bit more, without us having to look into how you switched designs

**Comment by fokkonaut** [comment]
I am just sending a map change, including map details. I should probably stop snapping and sending other messages during loading and downloading.
I have never used asan, sorry :/

**Comment by Robyt3** [comment]
> I am just sending a map change, including map details. I should probably stop snapping and sending other messages during loading and downloading.

Yes, you should. The client can't handle snapshots before the map is loaded. Crash can be reproduced by sending snapshots and other messages before the map change.

**Comment by Robyt3** [comment]
This is still a relevant issue.
